doc/sphinx/Makefile.am

```diff
@@ -159,14 +162,22 @@ if BUILD_SPHINX_DOCS
 
 all-local:
 	@for book in $(BOOKS); do					\
+		case "$$book" in \
+			Python_API) formats="html" ;; \
+			*) formats=$(BOOK_FORMATS) ;; \
+		esac; \
 		$(MAKE) $(AM_MAKEFLAGS) BOOK=$$book			\
 			PAPER="$(PAPER)" SPHINXFLAGS="$(SPHINXFLAGS)"	\
-			BOOK_FORMATS="$(BOOK_FORMATS)" $$book/_build;	\
+			BOOK_FORMATS="$$formats" $$book/_build;	\
```

Comment:
I think it's still useful to let the user override the formats (e.g. to create PDF and EPUB). I'm inclined to leave it like the rest and let singlehtml be broken; the user can specify BOOK_FORMATS to change it, and hopefully sphinx will one day fix it. Alternatively separate PYTHON_BOOKS/PYTHON_BOOK_FORMATS variables, or setting formats to BOOK_FORMATS with s/singlehtml/html/, would be fine.

Reply:
Fixed.  It looks like we also set BOOK_FORMATS to something sensible in doc/Makefile.am (I was just looking at doc/sphinx/Makefile.am where we set it to singlehtml only), so we're fine here.